lib/HLSL/DxilModule.cpp

```diff
@@ -291,6 +291,28 @@ unsigned DxilModule::GetGlobalFlags() const {
   return Flags;
 }
 
+static bool IsResourceSingleType(llvm::Type *Ty) {
```

Comment:
Hate to do this at this point, but this function should probably be called IsResourceSingleComponent, rather than SingleType.